CRM/MembershipExtras/BAO/MembershipPeriod.php

```diff
@@ -269,30 +266,66 @@ private static function doesOverlapWithOtherActivePeriods($periodParams) {
     }
 
     if (!$periodIsActive) {
-      return FALSE;
+      return;
     }
 
     if (empty($periodNewStartDate)) {
       throw new CRM_Core_Exception("Can't calculate period overlapping without period start date!");
     }
 
+    $overlappedMembershipPeriods = self::getOverlappingPeriods($periodID, $membershipID, $periodNewStartDate, $periodNewEndDate);
+    if ($overlappedMembershipPeriods->N > 0) {
+      $overlappingStartDate = new DateTime($periodNewStartDate);
+      $overlappingEndDate = new DateTime($periodNewEndDate);
+      $conflictingPeriodStartDate = new DateTime($overlappedMembershipPeriods->start_date);
+      $conflictingPeriodEndDate = new DateTime($overlappedMembershipPeriods->end_date);
+      
+      $isEndDateOverlap = $overlappingEndDate > $conflictingPeriodStartDate && $overlappingStartDate < $conflictingPeriodStartDate;
```

Comment:
regarding : 
```
 $isEndDateOverlap = $overlappingEndDate > $conflictingPeriodStartDate && $overlappingStartDate < $conflictingPeriodStartDate;
```

I think 

```
$isEndDateOverlap = $overlappingEndDate >= $conflictingPeriodStartDate
```

is enough

Reply:
I left checking of overlapping as generic as possible, so we can use it later on other parts if we need to. So basically, a start date overlaps if it is between the start and end dates of another period. And and end date overlaps if it is within the start and end dates of another period. Taking into account periods can have an empty end date.

Reply:
I could be wrong but a period could still overlap even if the start or end date is not within the start dates of another period if it starts before and ends after another period. Just FYI.

Reply:
make sense @MiyaNoctem  but just a small correction regarding
> a start date overlaps if it is between the start and end dates of another period

this should not be just  **between**, but rather **between or equal**


@jamienovick that's correct and we already talked about it in another comment above.